go_covert_lib/controller/processor/symmetricEncryption/symmetricEncryption.go

```diff
@@ -0,0 +1,237 @@
+package symmetricEncryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/des"
+	"errors"
+)
+
+// padding is done by having 7 zero's at the start followed by the length of 
+// the original message, the contents of the message, followed by a series of
+// zero's to fill the block size requirements of the symmetric encryption 
+// algorithm 
+const padIndex = 7
+
+type SymmetricEncryption struct {
+	algorithm string
+	mode      string
+	key       []byte
+}
+
+func (c *SymmetricEncryption) Process(data []byte) ([]byte, error) {
+	data = Pad(data)
+	algorithm := c.algorithm
+
+	var (
+		blockSize int
+		block     cipher.Block
+		err       error
+	)
+
+	// based on the users choice of symmetric algorithm create a cipher
+	switch algorithm {
+	case "Advanced Encryption Standard (AES)":
+		block, err = aes.NewCipher(c.key)
+		blockSize = 16
+	case "Data Encryption Standard (DES)":
+		block, err = des.NewCipher(c.key)
+		blockSize = 8
+	case "Triple Data Encryption Standard (3DES)":
+		block, err = des.NewTripleDESCipher(c.key)
+		blockSize = 8
+	default:
+		return nil, errors.New("Undefined algorithm selected")
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	// based on the users choice of the mode of operation encrypt in that mode
+	var cipherText []byte
+	selectedMode := c.mode
+	switch selectedMode {
+	case "Galois Counter Mode (GCM)":
+		cipherText = GCMEncrypter(block, data)
+		if cipherText == nil {
+			return nil, errors.New("Unable to encrypt in Galosis 
+									Counter Mode (GCM)")
+		}
+	case "Cipher Block Chaining (CBC)":
+		cipherText = CBCEncrypter(block, data, blockSize)
+	case "Cipher Feedback (CFB)":
+		cipherText = CFBEncrypter(block, data, blockSize)
+	case "Counter (CTR)":
+		cipherText = CTREncrypter(block, data, blockSize)
+	default:
+		return nil, errors.New("Undefined mode selected")
+	}
+
+	return cipherText[:], nil
+}
+
```

Comment:
These functions should return errors along with the byte array. The way you have set it up if the block fails:
1) The error message is lost, because it is never returns, so the user can't know what caused the failure..
2) You return nil from the encrypter as the byte array, so the caller won't realise that an error occurred because it will look at the err returned from the Process function and see it is nil, so assume everything is fine. The likely outcome of this is that when the covert channel attempts to send the bytes it will treat it as though it sent an empty string, so the user will think that everything is fine and the message was successfully sent.